### PR TITLE
Fix: add process context in workflow

### LINKED ISSUE #3276: WorkflowStepDefinition can not use “context.appName，context.appRevision...”
**Describe the bug**
WorkflowStepDefinition can not use “context.appName，context.appRevision...”

**Expected behavior**
can use Context Variable in WorkflowStepDefinition

**Screenshots**
image

image

**KubeVela Version**
1.2.1

### PR DESCRIPTION
**Pull request by FogDong**
Signed-off-by: FogDong <dongtianxin.tx@alibaba-inc.com>

### Description of your changes

1. Add process context in workflow
2. Add `workflowName` & `publishVersion` in process context.

Fixes #3276 

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### Special notes for your reviewer

/cc @barnettZQG

### CODE REVIEW

File: pkg/cue/process/handle.go [hunk 1]
```diff
@@ -42,6 +43,8 @@ type Context interface {
 	PushData(key string, data interface{})
 	GetCtx() context.Context
 	SetCtx(context.Context)
+	SetHooks(baseHooks []BaseHook, auxHooks []AuxiliaryHook)
+	SetComponents(components []common.ApplicationComponent)
```
Comment (wonderflow) [inline pkg/cue/process/handle.go:47]: Will it be better to use PushData ?

File: pkg/cue/process/handle.go [hunk 2]
```diff
@@ -99,44 +97,63 @@ type RequiredSecrets struct {
...
+// 	pCtx.SetHooks(baseHooks, auxHooks)
+// 	return pCtx
+// }
+
+// // NewPolicyContext create Application Scope templateContext for Policy
+// func NewPolicyContext(namespace, name, appName, appRevision string, components []common.ApplicationComponent, anno map[string]string) Context {
+// 	pCtx := NewContext(namespace, name, appName, appRevision, anno)
+// 	pCtx.PushData(model.ContextComponents, components)
+// 	return pCtx
+// }
```
Comment (wonderflow) [inline pkg/cue/process/handle.go:156]: let's delete them if not used any more.

File: pkg/cue/process/handle.go [hunk 3]
```diff
@@ -183,11 +200,6 @@ func (ctx *templateContext) BaseContextFile() string {
 		buff += fmt.Sprintf(model.OutputFieldName+": %s\n", structMarshal(ctx.base.String()))
 	}
 
-	if ctx.components != nil {
-		bt, _ := json.Marshal(ctx.components)
-		buff += fmt.Sprintf(model.ContextComponents+":%s\n", string(bt))
-	}
```
Comment (wonderflow) [inline pkg/cue/process/handle.go:196]: Have we checked the equivalency?

**Review by wonderflow — APPROVED** [review]
LGTM

File: pkg/appfile/appfile.go [hunk 4]
```diff
@@ -849,22 +851,39 @@ func generateComponentFromHelmModule(wl *Workload, appName, revision, ns string)
...
+	rls, repo, err := helm.RenderHelmReleaseAndHelmRepo(wl.FullTemplate.Helm, wl.Name, ctxData.AppName, ctxData.Namespace, wl.Params)
 	if err != nil {
 		return nil, err
 	}
 	compManifest.PackagedWorkloadResources = []*unstructured.Unstructured{rls, repo}
 	return compManifest, nil
 }
+
+// GenerateContextDataWithCtx generates process context data with context and appfile
+func GenerateContextDataWithCtx(ctx context.Context, appfile *Appfile, wlName string) process.ContextData {
```
Comment (zzxwill) [inline pkg/appfile/appfile.go:875]: Why do we need `wlName` for generating workflowName and PublishVersion?

Comment (FogDong) [reply pkg/appfile/appfile.go]: It is used for generating process context data, in which `wlName` will be `context.name`.

**Review by zzxwill — APPROVED** [review]

File: pkg/workflow/tasks/custom/task.go [hunk 5]
```diff
@@ -237,6 +238,9 @@ func (t *TaskLoader) makeValue(ctx wfContext.Context, templ string, id string) (
 		}
 		contextTempl = fmt.Sprintf("\ncontext: {%s}\ncontext: stepSessionID: \"%s\"", ms, id)
 	}
+	contextTempl += "\n" + pCtx.ExtendedContextFile()
+
+	fmt.Println("=================", contextTempl)
```
Comment (leejanee) [inline pkg/workflow/tasks/custom/task.go]: remove debug log

**Review by hongchaodeng — APPROVED** [review]

**Comment by hongchaodeng** [comment]
@captainroy-hy Please help review and merge this PR.

**Review by captainroy-hy — APPROVED** [review]

**Review by leejanee — APPROVED** [review]

**Pull request merged**